gratipay/models/participant.py

```diff
@@ -631,29 +635,62 @@ def remove_email(self, address):
             c.run("DELETE FROM emails WHERE participant=%s AND address=%s",
                   (self.username, address))
 
-    def send_email(self, spt_name, accept_lang=None, **context):
+    def _render_email(self, spt_name, accept_lang=None, **context):
         context['escape'] = escape
         context['username'] = self.username
         context.setdefault('include_unsubscribe', True)
-        email = context.setdefault('email', self.email_address)
+        context.setdefault('email', self.email_address)
         langs = i18n.parse_accept_lang(accept_lang or self.email_lang or 'en')
         locale = i18n.match_lang(langs)
         i18n.add_helpers_to_context(self._tell_sentry, context, locale)
         spt = self._emails[spt_name]
         base_spt = self._emails['base']
+        subject = spt['subject'].render(context)
         def render(t):
             b = base_spt[t].render(context).strip()
             return b.replace('$body', spt[t].render(context).strip())
-        message = {}
-        message['from_email'] = 'support@gratipay.com'
-        message['from_name'] = 'Gratipay Support'
-        message['to'] = [{'email': email, 'name': self.username}]
-        message['subject'] = spt['subject'].render(context)
-        message['html'] = render('text/html')
-        message['text'] = render('text/plain')
+        result = {
+            'from_email': 'support@gratipay.com',
+            'from_name': 'Gratipay Support',
+            'subject': subject,
+            'html': render('text/html'),
+            'text': render('text/plain')
+        }
+        return result
+
+    def send_email(self, spt_name, accept_lang=None, **context):
+        email_address = context.get('email', self.email_address)
+        message = self._render_email(spt_name, accept_lang, **context)
+        message['to'] = [{'email': email_address, 'name': self.username}]
 
         return self._mailer.messages.send(message=message)
 
+    @classmethod
+    def send_emails_to(cls, participants, spt_name, check_for=None, **context):
+        for p in participants:
+            if not p.email_address:
+                continue
+            send = getattr(p, check_for) if check_for else True
+            if send:
+                p.send_email(spt_name, **context)
+
+    def notify_patrons(self, patrons=None, elsewhere=None):
+        if patrons is None:
+            patrons = self.get_patrons()
+        if elsewhere is None:
+            elsewhere = self.get_unclaimed_elsewhere()
+        threading.Thread(
+            target=self.send_emails_to,
+            name='email',
+            args=(patrons, 'notify_patron'),
+            kwargs={
+                'check_for': 'notify_on_opt_in',
+                'elsewhere_username': elsewhere.user_name,
+                'elsewhere_platform': elsewhere.platform,
+                'profile_url': self.profile_url
+            }
+        ).start()
```

Comment:
I don't think that's a good idea, if we need an email thread then there should only be one, and it would be better if the queue was persistent so notifications don't get lost if the app crashes.